prow/config/config.go

```diff
@@ -174,9 +174,32 @@ type Plank struct {
 	// DefaultDecorationConfig are defaults for shared fields for ProwJobs
 	// that request to have their PodSpecs decorated
 	DefaultDecorationConfig *prowapi.DecorationConfig `json:"default_decoration_config,omitempty"`
+	// Deprecated, use JobURLPrefixConfig instead
 	// JobURLPrefix is the host and path prefix under
 	// which job details will be viewable
+	// TODO @alvaroaleman: Remove in September 2019
 	JobURLPrefix string `json:"job_url_prefix,omitempty"`
+	// JobURLPrefixConfig is the host and path prefix under which job details
+	// will be viewable. Use `*` as key for your global default.
+	JobURLPrefixConfig map[string]JobURLPrefixConfig `json:"job_url_prefix_config,omitempty"`
+}
+
+type JobURLPrefixConfig struct {
+	URL   string            `json:"url,omitempty"`
+	Repos map[string]string `json:"repos,omitempty"`
+}
+
+func (p Plank) GetJobURLPrefix(refs *prowapi.Refs) string {
+	if refs == nil {
+		return p.JobURLPrefixConfig["*"].URL
+	}
+	if p.JobURLPrefixConfig[refs.Org].Repos[refs.Repo] != "" {
+		return p.JobURLPrefixConfig[refs.Org].Repos[refs.Repo]
+	}
+	if p.JobURLPrefixConfig[refs.Org].URL != "" {
+		return p.JobURLPrefixConfig[refs.Org].URL
+	}
+	return p.JobURLPrefixConfig["*"].URL
```

Comment:
Instead of using multiple levels of maps and defaults, I think it would be preferable to use the pattern that we use elsewhere (especially for consistency): use a `map[string]string` where the keys are either `org` or `org/repo` format and check for values in the following order `org/repo` > `org` > `*`.

Reply:
That is definitely simpler but we are totally inconsistent as to when it does or does not allow for an override, which the tiered way does. #9757 

/shrug

Reply:
/shrug

Reply:
True. Ironically I've thought about doing it like that initially but ditched the idea because there is no safe way to distinguish repos from orgs. But yeah, enforcing to prefix repos with `org/` is the obvious solution for that.

I've changed it, IMHO makes both the resulting config and the tests much easier to read. LMK what you think.